app/assets/scripts/components/slate/plugins/equation/index.js

```diff
@@ -42,22 +54,35 @@ const deleteEquation = (editor) => {
  *
  * @param {Editor} editor Slate editor instance.
  */
-export const upsertEquation = (editor, equation, nodePath) => {
+export const upsertEquation = (editor, equation, isInline, nodePath) => {
   const node = {
-    type: EQUATION,
+    type: isInline ? EQUATION_INLINE : EQUATION,
     children: [{ text: equation }]
   };
 
   if (nodePath) {
-    Transforms.removeNodes(editor, { at: nodePath });
+    // replace existing equation, we first add the updated equation and
+    // then remove the outdated one
     Transforms.insertNodes(editor, node, { at: nodePath });
+
+    // Adding a new inline node adds an empty text node after the equation, we need to
+    // remove the node after that.
+    const nodeOffset = isInline ? 2 : 1;
+    const removeElementIndex = nodePath[nodePath.length - 1] + nodeOffset;
+    const parentPath = nodePath.slice(0, -1);
+    Transforms.removeNodes(editor, { at: [...parentPath, removeElementIndex] });
```

Comment:
The empty text node is expected and supposed to be there. It is needed so that the cursor has a place to go to after the equation, otherwise you'd not be able to add content after it.

Reply:
This is also a slate constraint. See item no.4: https://docs.slatejs.org/concepts/11-normalizing